src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/NuGetTargetFrameworkMonikerComparer.cs

```diff
@@ -67,6 +72,85 @@ public int Compare(TargetFrameworkMoniker? x, TargetFrameworkMoniker? y)
             };
         }
 
+        public TargetFrameworkMoniker Merge(TargetFrameworkMoniker tfm1, TargetFrameworkMoniker tfm2)
+        {
+            if (tfm1 is null)
+            {
+                throw new ArgumentNullException(nameof(tfm1));
+            }
+
+            if (tfm2 is null)
+            {
+                throw new ArgumentNullException(nameof(tfm2));
+            }
+
+            var nugetTfm1 = NuGetFramework.Parse(tfm1.Name);
+            var nugetTfm2 = NuGetFramework.Parse(tfm2.Name);
+
+            // We can only combine if the platform is the same or at least one is null.
+            if (!IsPlatformSameOrNull(nugetTfm1, nugetTfm2))
+            {
+                throw new InvalidOperationException("Cannot merge two target frameworks with different platforms");
```

Comment:
This is a bit picky because this isn't likely to come up often, but if a filter did try to apply a TFM with a different platform from the current one, we wouldn't want it to crash. We'd want the merge to fail and alert the user that the TFM we're using might not be right. So, it may be better to have this method return a bool instead of throwing in this case.

Reply:
Good idea. Add a log message instead